### PR TITLE
Add filesystem integration

### PR DESCRIPTION
**Pull request by RaoulMeyer**
### Description

This adds an integration for the native `file_get_contents` and `file_put_contents` functions. In a follow up, I want to also add spans to `fopen`/`fread`/`fwrite`/`fclose`. 

### Readiness checklist
- [X] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the appropriate release draft. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

**Comment by RaoulMeyer** [comment]
Looks like a lot of framework tests now fail, because they do filesystem things and assert on an exact list of spans. What is the desired way to fix this? Seems to me like integration tests should not be affected by changes in other integrations, but that might be intended.

**Comment by SammyK** [comment]
Thanks for the integration PR @RaoulMeyer! I'm really happy you tackled some of the core IO functions. We'll get this reviewed this week and get back with you on any feedback. :)

**Comment by SammyK** [comment]
Thanks again for the fantastic integration PR, @RaoulMeyer!

There are two few points I'd like to get @labbati and @morrisonlevi's feedback on:

0. For PHP standard lib integrations - should we prefix the integration names with `PhpStd` or something similar?
1. @RaoulMeyer had an excellent point about this affecting the other integrations. Do we want to have this integration disabled by default or perhaps just explicitly disable it for the integration tests that have file I/O operations? Or some other idea?

@RaoulMeyer: We'll need a little bit of extra time to sort the implementation details. Until then, since we've been doing a lot of merging to master recently, there's a conflict on this PR now. Would you mind resolving that? :)

**Comment by RaoulMeyer** [comment]
@SammyK Thanks! I rebased the PR, as soon as you've decided on the details I'll this PR update accordingly.

**Comment by morrisonlevi** [comment]
Once we merge in https://github.com/DataDog/dd-trace-php/pull/585 then we can "assert a hierarchical graph of spans with parent child relations", which may be the way we can make things like this more robust in the future. I haven't thoroughly reviewed the PR to see if how it handles extra spans, though, so maybe that won't work out.

File: src/DDTrace/Type.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 class Type
 {
+    const FILESYSTEM = 'filesystem';
```
Comment (morrisonlevi) [inline src/DDTrace/Type.php]: Given that these functions can do more than filesystems, I think we should not name it 'filesystem'. I've been thinking more about it; these are really stream operations. Maybe we call it 'phpstream' or something?

**Comment by krakjoe** [comment]
While enthusiasm was initially high, this no longer fits into our road map and the conversation went stale long ago.

We're grateful for the time you spent here, and we apologize for the long delay in feedback, but I'm going to close this PR as we won't be moving forward in this direction.

Thanks again.

**Pull request closed without merging**